src/App.tsx

```diff
@@ -1,204 +1,104 @@
-/* eslint-disable jsx-a11y/label-has-associated-control */
-/* eslint-disable jsx-a11y/control-has-associated-label */
-import React from 'react';
-import { UserWarning } from './UserWarning';
-import { USER_ID } from './api/todos';
+import React, { useState, useEffect, useCallback, useMemo } from 'react';
+import * as todosService from './api/todos';
+
+import { ErrorMessages } from './components/ErrorsMessage';
+import { TodoList } from './components/TodoList';
+import { Footer } from './components/Footer';
+import { Header } from './components/Header';
+
+import { Todo } from './types/Todo';
+import { Field } from './types/Field';
 
 export const App: React.FC = () => {
-  if (!USER_ID) {
-    return <UserWarning />;
-  }
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [todosCounter, setTodosCounter] = useState<number>(0);
+  const [errorMessage, setErrorMessage] = useState('');
+  const [field, setField] = useState<Field>(Field.ALL);
+
+  useEffect(() => {
+    todosService
+      .getTodos()
+      .then(setTodos)
+      .catch(() => setErrorMessage('Unable to load todos'));
+  }, []);
+
+  useEffect(() => {
+    const timerId = window.setTimeout(() => setErrorMessage(''), 4000);
+
+    return () => clearTimeout(timerId);
+  }, [errorMessage]);
+
+  useEffect(() => {
+    const todosActive = todos.filter(todo => !todo.completed);
+
+    setTodosCounter(todosActive.length);
+  }, [todos]);
+
+  const changeComplete = useCallback(
```

Comment:
You use `useCallback` but don't use `React.memo`
In this case `useCallback` is not really needed here

Reply:
Я думала, що якщо я передаю функцію в props, то завжди використовую useCallback

Reply:
Using `useCalback` makes sense when you wrap your child component in `React.memo`. Otherwise, your child component will rerender in any case, and using `useCallback` doesn't have any effect